src/torrent.ts

```diff
@@ -201,15 +206,40 @@ export async function getTorrentByFuzzyName(
 	name: string
 ): Promise<null | Metafile> {
 	const allNames = await db("torrent").select("name", "file_path");
-	// @ts-expect-error fuse types are confused
-	const potentialMatches = new Fuse(allNames, {
-		keys: ["name"],
-		distance: 6,
-		threshold: 0.25,
-	}).search(name);
-
-	if (potentialMatches.length === 0) return null;
-	const [firstMatch] = potentialMatches;
+
+	const fullMatch = reformatTitleForSearching(name).replace(
+		/[^a-z0-9]/gi,
+		""
+	);
+
+	// Attempt to filter torrents in DB to match incoming torrent before fuzzy check
+	let filteredNames = [];
+
+	if (fullMatch) {
+		filteredNames = allNames.filter((dbName) => {
+			const dbMatch = reformatTitleForSearching(dbName.item.name).replace(
+				/[^a-z0-9]/gi,
+				""
+			);
```

Comment:
maybe for the future: I wonder if we would only want to check season and episode number strictly and then leave the name matching to the fuzzy search?

Reply:
I think this is reasonable, especially if ufuzzy is more consistent than fuse. We can see how this goes and if it misses a lot of matches maybe dial back the prefiltering

Reply:
you can drag/drop a json or text list into the demo to compare match quality between Fuse and uFuzzy: https://leeoniya.github.io/uFuzzy/demos/compare.html?libs=uFuzzy,fuzzysort,QuickScore,Fuse&search=super%20ma